File: app/src/main/java/net/veldor/flibustaloader/database/dao/BookmarksDao.java
package net.veldor.flibustaloader.database.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import net.veldor.flibustaloader.database.entity.Bookmark;

import java.util.List;

@Dao
public interface BookmarksDao {

    @Query("SELECT * FROM Bookmark WHERE id = :id")
    Bookmark getBookmarkById(String id);

    @Query("SELECT * FROM Bookmark WHERE id = :id")
    Bookmark getBookmarkById(int id);


    @Query("SELECT * FROM Bookmark ORDER BY id DESC")
    List<Bookmark> getAllBookmarks();

    @Insert
    void insert(Bookmark book);

    @Delete
    void delete(Bookmark book);

    @Query("SELECT * FROM Bookmark WHERE name = :name AND link = :link")
    List<Bookmark> getDuplicate(String name, String link);

/*    @Update
    void update(ReadedBooks book);
*/
}

File: app/src/main/java/net/veldor/flibustaloader/database/dao/BookmarksDao.kt
package net.veldor.flibustaloader.database.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import net.veldor.flibustaloader.database.entity.Bookmark

@Dao
interface BookmarksDao {
    @Query("SELECT * FROM Bookmark WHERE id = :id")
    fun getBookmarkById(id: String?): Bookmark?

    @Query("SELECT * FROM Bookmark WHERE id = :id")
    fun getBookmarkById(id: Int): Bookmark?

    @get:Query("SELECT * FROM Bookmark ORDER BY id DESC")
    val allBookmarks: MutableList<Bookmark>

    @Insert
    fun insert(book: Bookmark?)

    @Delete
    fun delete(book: Bookmark?)

    @Query("SELECT * FROM Bookmark WHERE name = :name AND link = :link")
    fun getDuplicate(name: String?, link: String?): List<Bookmark?>? /*    @Update
    void update(ReadedBooks book);
*/
}